hawkauthlib/__init__.py

```diff
@@ -178,16 +174,26 @@ def verify_payload(request, params=None, algorithm=None):
     if params is None:
         params = utils.parse_authz_header(request, {})
 
-    return utils.strings_differ(params["hash"], hash_payload(request, params, algorithm))
+    return not utils.strings_differ(params["hash"], hash_payload(request, params, algorithm))
 
 
 @utils.normalize_request_object
-def check_signature(request, key, hashmod=None, params=None, nonces=None):
+def check_signature(request, key, require_payload_hash=False, verify_payload_hash=True, algorithm=None, params=None, nonces=None):
```

Comment:
There are two choices that the server might need to make, hence two parameters here:

* Do we reject signatures that do not include a payload hash?
* Do we check the payload hash if provided?

Reply:
> Do we check the payload hash if provided?
Not a decision, the Signature **must** include the hash attribute if provided or signatures will not match